### PR TITLE
Ensure spam doesn't hit logged-out /latest

### LINKED ISSUE #18979: Ensure spam doesn’t hit logged-out /latest
## User Story (Who / What / Why)

As the admin of a Forem, I want to ensure that low quality content is not indexed in Google to avoid SEO penalty.  This motivation has spurred on work for other related projects — most notably hiding comments from the user profile view for logged-out visitors.

Similar to how we recently adjusted the “score count” thresholds and number of comments that show up *for signed out traffic*, we want to adjust the score count for `/latest` *for anonymous traffic*. 

To generate consistency, we want to re-use the `home_feed_minimum_score` as the default score for `latest` in this context — while keeping the same default present for *signed in* traffic********.********

We will put a notice at the top of /latest for signed out traffic, similar to what we have for all tabs for tag pages:

Image

This should read:

Some **latest** posts are only visible for members. [[Sign in](https://dev.to/enter?state=new-user)](https://dev.to/enter?state=new-user) to see all latest.

## Definition of Done

- [ ]  Change minimum score for signed out contexts (`app/services/articles/feeds/latest.rb`)
- [ ]  Add notice at top of `/latest` as described above.

### PR DESCRIPTION
**Pull request by jaw6**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

We want to suppress low-quality content from search engines. The `/latest` list of articles should use `home_feed_minimum_score` instead of a hard-coded low score.

We also want to include a message about the suppressed content, prompting visitors to sign-in:

Screen Shot 2023-02-13 at 11 40 26

## Related Tickets & Documents

- Closes #18979

## QA Instructions, Screenshots, Recordings

Note that the `home_feed_minimum_score` scope also includes `featured` articles (this _might_ be an unanticipated change, there's some discussion of this scope in [this PR](https://github.com/forem/forem/pull/15850)).

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [x] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by jaw6** [comment]
Articles::Feeds::Latest before this PR:

```
Article Load (1.7ms)  SELECT "articles"."path",
"articles"."title",
"articles"."id",
"articles"."published",
"articles"."comments_count",
"articles"."public_reactions_count",
"articles"."cached_tag_list",
"articles"."main_image",
"articles"."main_image_background_hex_color",
"articles"."updated_at",
"articles"."slug",
"articles"."video",
"articles"."user_id",
"articles"."organization_id",
"articles"."video_source_url",
"articles"."video_code",
"articles"."video_thumbnail_url",
"articles"."video_closed_caption_track_url",
"articles"."experience_level_rating",
"articles"."experience_level_rating_distribution",
"articles"."cached_user",
"articles"."cached_organization",
"articles"."published_at",
"articles"."crossposted_at",
"articles"."description",
"articles"."reading_time",
"articles"."video_duration_in_seconds",
"articles"."last_comment_at" FROM "articles" WHERE "articles"."published" = $1 AND (published_at <= '2023-02-13 10:06:13.185516') AND (score > -20) AND "articles"."main_image" IS NOT NULL ORDER BY "articles"."published_at" DESC LIMIT $2 OFFSET $3  
[["published",true],["LIMIT",1],["OFFSET",0]]
```

After this PR:
```
Article Load (1.2ms)  SELECT "articles"."path",
"articles"."title",
"articles"."id",
"articles"."published",
"articles"."comments_count",
"articles"."public_reactions_count",
"articles"."cached_tag_list",
"articles"."main_image",
"articles"."main_image_background_hex_color",
"articles"."updated_at",
"articles"."slug",
"articles"."video",
"articles"."user_id",
"articles"."organization_id",
"articles"."video_source_url",
"articles"."video_code",
"articles"."video_thumbnail_url",
"articles"."video_closed_caption_track_url",
"articles"."experience_level_rating",
"articles"."experience_level_rating_distribution",
"articles"."cached_user",
"articles"."cached_organization",
"articles"."published_at",
"articles"."crossposted_at",
"articles"."description",
"articles"."reading_time",
"articles"."video_duration_in_seconds",
"articles"."last_comment_at" FROM "articles" WHERE "articles"."published" = $1 AND (published_at <= '2023-02-13 10:05:07.855410') AND ("articles"."featured" = $2 OR "articles"."score" >= $3) AND "articles"."main_image" IS NOT NULL ORDER BY "articles"."published_at" DESC LIMIT $4 OFFSET $5  [["published",true],["featured",true],["score",0],["LIMIT",1],["OFFSET",0]]
```

(I don't actually trust the time measure here, it's probably an accident of local db query caching -- there's no reason to think the new query should be any faster.)

File: app/views/stories/_main_stories_feed.html.erb [hunk 1]
```diff
@@ -1,3 +1,12 @@
+<% if should_show_latest_spam_suppression? %>
```
Comment (lightalloy) [inline app/views/stories/_main_stories_feed.html.erb:1]: What do you think about passing `stories` explicitly? No need to disable rubocop as a bonus.

**Comment by benhalpern** [comment]
I'm not sure this satisfies the behavior. I think this would alter the criteria for all users, when we only want it to change for anonymous users. Authenticated users should still see the `-20` threshold.

**Comment by jaw6** [comment]
> I'm not sure this satisfies the behavior. I think this would alter the criteria for all users, when we only want it to change for anonymous users. Authenticated users should still see the `-20` threshold.

@benhalpern That could be! I was having a little trouble reasoning about the code. But I had thought that [this `unless user_signed_in?`](https://github.com/forem/forem/blob/dbf766012af19c4f86e6cfd4a7c3b9b62bd23fec/app/controllers/stories_controller.rb#L127) meant that `Articles::Feeds::Latest` wasn't used for signed-in-users. I'll keep digging.

```ruby
 def handle_base_index
    @home_page = true
    assign_feed_stories unless user_signed_in? # Feed fetched async for signed-in users
```

```ruby
  def assign_feed_stories
    if params[:timeframe].in?(Timeframe::FILTER_TIMEFRAMES)
      @stories = Articles::Feeds::Timeframe.call(params[:timeframe])
    elsif params[:timeframe] == Timeframe::LATEST_TIMEFRAME
      @stories = Articles::Feeds::Latest.call
```

**Comment by benhalpern** [comment]
@jaw6 yeah, I may be wrong about that presumption. Thanks for looking into it and testing the scenaro. 👍

File: app/views/stories/_main_stories_feed.html.erb (hunk 1 above)
Comment (jaw6) [reply app/views/stories/_main_stories_feed.html.erb]: :+1:

**Review by rt4914 — COMMENTED** [review]
I am not fully sure of the changes in rb files and therefore I am assigning it to @benhalpern

**Review by benhalpern — APPROVED** [review]

**Comment by jaw6** [comment]
Closing & re-opening, hoping to trick CI into working properly.

**Pull request merged**